host_core/lib/host_core/structured_logger.ex

```diff
@@ -132,8 +132,12 @@ defmodule HostCore.StructuredLogger.FormatterJson do
            {time, m} <- Map.pop(m, :time),
            {registered_name, m} <- Map.pop(m, :registered_name) do
         m
-        |> put_val(:host_id, HostCore.Host.host_key())
-        |> put_val(:lattice_id, HostCore.Host.lattice_prefix())
+        # TODO: devise a reliable way we can glean the host and lattice
```

Comment:
Is this a blocker for releasing?

Reply:
The only side effect of not doing this now is that between now and when we find a solution, structured JSON logs will not contain the host ID field or the lattice prefix field. The interesting thing is that under the new supervision tree, there are many legitimate cases that don't have a host context (e.g. stuff emitted from the lattice managers or the other utility/caching tools).

Put another way, even after we figure this out, not all structured log statements will contain a host and/or lattice ID

Reply:
Whats bugs me about the old way of doing this is that both `host_key()` and `lattice_prefix()` used to push a request through the singleton host's mailbox. If we add some kind of metadata where we do `get_host(pid)` and `get_lattice_prefix(pid)`, we're still going to incur that cost. Tradeoff is that it would be no slower than before.

Reply:
ok, found a remarkably easy solution to this... You'll see these values like this:
```
{"app":"host_core","host_id":"ND34K3PYPJVGVG33TY5KTIXFBN7MNS7SQ33364FWZRH7RK2ODWBZQ7NG","lattice_prefix":"default","level":"info","mfa":"HostCore.Vhost.VirtualHost.publish_heartbeat/1","pid":"#PID<0.535.0>","time":"2022-11-28T14:16:02.421491Z","msg":"Generating heartbeat"}
```

Reply:
`Logger.metadata` adds metadata that will be associated with log output for the entire lifetime of the process. So we issue a logger.metadata during the init of actors and providers and the lattice supervisor and now we've got all the info we need _without_ making a round trip through a process mailbox. 👨‍🍳 💋